test/common/json/json_fuzz_test.cc

```diff
@@ -23,6 +23,16 @@ DEFINE_FUZZER(const uint8_t* buf, size_t len) {
     ProtobufWkt::Struct message2;
     MessageUtil::loadFromJson(MessageUtil::getJsonStringFromMessage(message), message2);
     FUZZ_ASSERT(TestUtility::protoEqual(message, message2));
+
+    // MessageUtil::getYamlStringFromMessage automatically convert types, so we have to do another
+    // round-trip.
+    std::string yaml = MessageUtil::getYamlStringFromMessage(message);
```

Comment:
Do you know if we can use `check_format` to limit the use of YAML parsing? Ideally we ensure that it's only ever used on the control plane, and most likely only for files coming locally or from CLI args. If we can get a restriction like this, we can move YAML parsing from security consideration structurally.

Reply:
Yeah we should do that, but thats out of scope of this PR. I'll explore this later. @jmarantz any quick idea?

Reply:
Disallow them generally and whitelist where we have OK'd them, if that's not too onerous.

This pattern is used in a few places and isn't too bad I think.

Reply:
SGTM, let me take that in another PR. This one is pure re-implementation.

Reply:
@lizan sure, we can do in another PR, can you file a tracking issue?

Reply:
https://github.com/envoyproxy/envoy/issues/9039